src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -295,7 +300,27 @@ private void perform(Run<?,?> build, Launcher launcher, TaskListener listener) t
     	final IProgress progress = new ScanProgress(listener);
     	final boolean suspend = m_wait;
     	final IScan scan = ScanFactory.createScan(getScanProperties(build, listener), progress, m_authProvider);
-        
+		JenkinsAuthenticationProvider checkAppScan360Connection = new JenkinsAuthenticationProvider(m_credentials,build.getParent().getParent());
```

Comment:
Why are we creating another instance of JenkinsAuthenticationProvider instead of just using m_authProvider that was just initialized a few lines before this?

Reply:
Matt, I created this to call the isAppScan360() method which is in JenkinsAuthenticationProvider class.

Reply:
A few lines up on line 294 we're intializing m_authProvider with the same call to the JenkinsAuthenticationProvider constructor. So why do we need another instance of JenkinsAuthenticationProvider? We should be able to use m_authProvider.

Reply:
Matt, m_authProvider had been declared as the global variable of the IAuthenticationProvider type. So, unable to call the isAppScan360() method using that instance.